pkg/segment/writer/rawchecker.go

```diff
@@ -52,10 +52,18 @@ func ApplySearchToMatchFilterRawCsg(match *MatchFilter, col []byte) (bool, error
 	if match.MatchOperator == And {
 		var foundQword bool = true
 		if match.MatchType == MATCH_PHRASE {
-			foundQword = utils.IsSubWordPresent(col[idx:idx+clen], match.MatchPhrase)
+			regexp := match.Regexp
+			if regexp == nil {
```

Comment:
nit: no need for an extra regexp variable, you can just use the `match.Regexp` as is